src/extra/imgui.hpp

```diff
@@ -35,5 +37,85 @@ struct Context {
   //   context = ::ImGui::CreateContext();
   // }
 };
+
+struct Enums {
+#define REGISTER_ENUM(_NAME_, _CC_)                                            \
+  static constexpr uint32_t _NAME_##CC = _CC_;                                 \
+  static inline EnumInfo<_NAME_> _NAME_##EnumInfo{#_NAME_, CoreCC,             \
+                                                  _NAME_##CC};                 \
+  static inline Type _NAME_##Type = Type::Enum(CoreCC, _NAME_##CC)
+
+#define REGISTER_FLAGS(_NAME_, _CC_)                                           \
+  static constexpr uint32_t _NAME_##CC = _CC_;                                 \
+  static inline FlagsInfo<_NAME_> _NAME_##EnumInfo{#_NAME_, CoreCC,            \
+                                                   _NAME_##CC};                \
+  static inline Type _NAME_##Type = Type::Enum(CoreCC, _NAME_##CC)
+
+#define REGISTER_FLAGS_EX(_NAME_, _CC_)                                        \
+  REGISTER_FLAGS(_NAME_, _CC_);                                                \
+  static inline Type _NAME_##SeqType = Type::SeqOf(_NAME_##Type);              \
+  static inline Type _NAME_##VarType = Type::VariableOf(_NAME_##Type);         \
+  static inline Type _NAME_##VarSeqType = Type::VariableOf(_NAME_##SeqType)
+
+  enum class GuiButton {
+    Normal,
+    Small,
+    Invisible,
+    ArrowLeft,
+    ArrowRight,
+    ArrowUp,
+    ArrowDown
+  };
+  REGISTER_ENUM(GuiButton, 'guiB'); // FourCC = 0x67756942
+
+  enum class GuiTableFlags {
+    None = ::ImGuiTableFlags_None,
+    Resizable = ::ImGuiTableFlags_Resizable,
+    Reorderable = ::ImGuiTableFlags_Reorderable,
+    Hideable = ::ImGuiTableFlags_Hideable,
+    Sortable = ::ImGuiTableFlags_Sortable
+  };
+  REGISTER_FLAGS_EX(GuiTableFlags, 'guiT'); // FourCC = 0x67756954
+
+  enum class GuiTableColumnFlags {
+    None = ::ImGuiTableColumnFlags_None,
+    Disabled = ::ImGuiTableColumnFlags_Disabled,
+    DefaultHide = ::ImGuiTableColumnFlags_DefaultHide,
+    DefaultSort = ::ImGuiTableColumnFlags_DefaultSort
+  };
+  REGISTER_FLAGS_EX(GuiTableColumnFlags, 'guTC'); // FourCC = 0x67755443
+
+  enum class GuiTreeNodeFlags {
+    None = ::ImGuiTreeNodeFlags_None,
+    Selected = ::ImGuiTreeNodeFlags_Selected,
+    Framed = ::ImGuiTreeNodeFlags_Framed,
+    OpenOnDoubleClick = ::ImGuiTreeNodeFlags_OpenOnDoubleClick,
+    OpenOnArrow = ::ImGuiTreeNodeFlags_OpenOnArrow,
+    Leaf = ::ImGuiTreeNodeFlags_Leaf,
+    Bullet = ::ImGuiTreeNodeFlags_Bullet,
+    SpanAvailWidth = ::ImGuiTreeNodeFlags_SpanAvailWidth,
+    SpanFullWidth = ::ImGuiTreeNodeFlags_SpanFullWidth
+  };
+  REGISTER_FLAGS_EX(GuiTreeNodeFlags, 'guTN'); // FourCC = 0x6775544E
+
+  template <typename EnumType> static EnumType getFlags(CBVar var) {
```

Comment:
Similarly, this could be moved alongside the macros.

However it does need to include the `magic_enum` header to work (because of  `bitwise_operators` namespace), so we need to carefully select in which file to put it.